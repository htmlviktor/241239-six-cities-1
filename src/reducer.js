import {loadOffers} from './actions';

const initialState = {
  currentCity: `Amsterdam`,
  listCities: new Set([...data.map((offer) => offer.city)]),
  offers: []
};

const Operation = {
  loadOffers: () => (dispatch, _getState, api) => {
    return api.get(`/hotels`)
      .then((response) => {
        dispatch(loadOffers(response.data));
      });
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `CITY_CHANGE`:
      return Object.assign({}, state, {
        currentCity: action.city
      });
    case `LOAD_OFFERS`:
      return Object.assign({}, state, {
        offers: action.offers
      });
    default: return state;
  }
};


export {
  reducer,
  Operation
};
