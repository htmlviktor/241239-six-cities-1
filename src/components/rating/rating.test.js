import React from 'react';
import renderer from 'react-test-renderer';
import Rating from './rating.jsx';
import {Router} from 'react-router-dom';
import history from '../../history';

import {createStore} from 'redux';
import {Provider} from 'react-redux';
import reducer from '../../reducer/index';


it(`Test Rating Component`, () => {
  const tree = renderer
    .create(
        <Provider store={createStore(reducer)}>
          <Router history={history}>
            <Rating rating={4}/>
          </Router>
        </Provider>
    )
      .toJSON();

  expect(tree).toMatchSnapshot();
});
