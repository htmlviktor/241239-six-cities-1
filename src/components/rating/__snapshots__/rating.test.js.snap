// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rating Component 1`] = `
<span
  style={
    Object {
      "width": "80%",
    }
  }
/>
`;
